File: src/main/java/guru/xgm/jass/openapi/actionSystem/convert/jass2angelscript/Jass2AngelScriptAction.java
package guru.xgm.jass.openapi.actionSystem.convert.jass2angelscript;

import guru.xgm.jass.openapi.actionSystem.convert.Jass2AnyAction;
import guru.xgm.jass.openapi.actionSystem.convert.Jass2AnyVisitor;
import org.jetbrains.annotations.NotNull;

public class Jass2AngelScriptAction extends Jass2AnyAction {

    public Jass2AngelScriptAction() {
        super();
    }

    @Override
    public @NotNull Jass2AnyVisitor getVisitor() {
        return new Jass2AngelScriptVisitor();
    }

    @Override
    public @NotNull String getTargetExtension() {
        return ".ass";
    }
}

File: src/main/kotlin/raft/war/language/jass/openapi/actionSystem/convert/jass2angelscript/Jass2AngelScriptAction.kt
package raft.war.language.jass.openapi.actionSystem.convert.jass2angelscript

import raft.war.language.angelscript.openapi.fileTypes.AngelScriptFileType
import raft.war.language.jass.openapi.actionSystem.convert.Jass2AnyAction
import raft.war.language.jass.openapi.actionSystem.convert.Jass2AnyVisitor

class Jass2AngelScriptAction : Jass2AnyAction() {
    override val visitor: Jass2AnyVisitor
        get() = Jass2AngelScriptVisitor()

    override val targetExtension: String
        get() = AngelScriptFileType.EXTENSION
}
